include/libdnf-cli/argument_parser.hpp

```diff
@@ -208,6 +210,10 @@ class ArgumentParser {
         // Returns the ArgumentParser instance to which the argument belongs.
         ArgumentParser & get_argument_parser() const noexcept { return owner; }
 
+        void set_user_data(ArgumentParserUserData * user_data) noexcept { this->user_data = user_data; }
+
+        ArgumentParserUserData * get_user_data() const noexcept { return user_data; }
```

Comment:
If I understand that correctly, the `get_user_data()` could be used later not only for getting the parent command as it is now in the `cli::session::Command` right?

Reply:
Thanks for the explanation, I got it now. I suppose when its called `ArgumentParserUserData`, it is something what is offered by the `ArgumentParser` to its clients. So I would expect the definition of this in this header as you mentioned. Do you see any cons in this approach?